DOCS-16527 replica set to sharded cluster page (#363)
mongodb/docs-cluster-to-cluster-sync: source/topologies/rs-to-sharded.txt

.. _c2c-rs-to-sharded:

=======================================
Sync a Replica Set to a Sharded Cluster
=======================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: twocols

> This page outlines the procedure to sync data between a replica set and a 

Reviewer: Since this is the first mention of the word "sync", I would use the full term (and then use sync for the rest one the page like you already have!)

```suggestion
This page outlines the procedure to synchronize data between a replica set and a 
```

Author: Good idea, changing now!

sharded cluster using :program:`mongosync`. To perform a sync between two

> replica sets, see :ref:`c2c-quickstart`.

Reviewer: [question] Does the source have to be a RS and the destination have to be a shared cluster? Or can it also go the other way around?

Author: It cannot be the other way around. I will clarify this.

Before You Begin
----------------

Before you begin your sync, define your source replica set, your 
destination sharded cluster, and an administrative user that has permissions in 

> both clusters. You must also download ``mongosync``. To learn more, see 

Reviewer: [suggestion] Since there are three things that a user needs to define, I would suggest putting those steps in a [list ](https://www.mongodb.com/docs/meta/style-guide/writing/write-clear-concise-sentences-paragraphs/#create-short-paragraphs)for better readability

> the :ref:`setup <c2c-quickstart-setup>` instructions.

Reviewer: [suggestion] for clarity

```suggestion
the :ref:`mongosync setup <c2c-quickstart-setup>` instructions.
```

Author: Fixing now!

Steps 
-----

.. procedure::
   :style: normal

   .. step:: Connect the replica set and sharded cluster. 

      ``mongosync`` must create an initial connection to the source and 
      destination clusters before you can start to sync data.
      Follow the :ref:`connection instructions <c2c-conn-top-level>` for your 
      cluster architecture to connect to the :binary:`mongos` instance in your 
      cluster.

      For an example of the ``mongosync`` connection command, see 

>       :ref:`Initialize mongosync <c2c-initialize>`.

Reviewer: [question] since the linked example of the mongosync connection command is fairly short, what do you think about including it here instead of requiring the user to click the link?

Reviewer: Alternatively, you could rephrase the sentence "For more information, see Initialize mongosync" to refer to more than just the example command.

Author: I am worried about this example because it is for a replica set to a replica set. In my mongosync behavior ticket, I created a more general example format for the command. I hope that gets merged first, and then I will link that example instead. I will leave this comment unresolved so that I can come back to it.

Author: Put in more generic code example so users will not have to click the link.

   .. step:: Start synchronization.

>        The :ref:`start <c2c-api-start>` endpoint initiates data 
>        synchronization. 

Reviewer: [suggestion] I would suggest making it clear here that the user has to take an action

```suggestion
       Call the :ref:`start <c2c-api-start>` endpoint initiates data 
       synchronization. 
```

Author: Fixing now!

       
       To sync from a replica set to a sharded cluster, set the 
       ``sharding`` option for the ``start`` command to shard collections on 
       the destination cluster. For more information, see 
       :ref:`Sharding Parameters <c2c-api-start-sharding>`.

>        The ``sharding.shardingEntries`` array specifies the collections to shard.

Reviewer: [suggestion] for clarity

```suggestion
       Use the ``sharding.shardingEntries`` parameter to specify the collections to shard.
```

Author: Fixing now!

>        Collections that are not listed in this array replicate as unsharded. For
>        more information, see :ref:`Shard Replica Sets <c2c-shard-replica-sets>`.

Reviewer: [suggestion] we can probably use active voice here!

```suggestion
       Collections that you do not list in this array replicate as unsharded. For
       more information, see :ref:`Shard Replica Sets <c2c-shard-replica-sets>`.
```

Author: Fixing now!

       The following example starts a sync from a replica set to a sharded
       cluster: 

       Request
       ~~~~~~~

       .. literalinclude:: /includes/api/requests/start-rs-shard.sh

       Response
       ~~~~~~~~

       .. literalinclude:: /includes/api/responses/success.json
          :language: json
          :copyable: false

Next Steps
----------

You can finalize a migration and transfer your application workload from the 
source to the destination cluster using the ``mongosync`` 
:ref:`cutover process <c2c-cutover-process>`.

Limitations
-----------

> - You cannot use the :ref:`reverse <c2c-api-reverse>` endpoint on a sync
>   between a replica set and a sharded cluster. 

Reviewer: [suggestion] for minimalism

```suggestion
- You cannot use the :ref:`reverse <c2c-api-reverse>` endpoint 
  between a replica set and a sharded cluster. 
```

.. _c2c-rs-to-sharded:

=======================================
Sync a Replica Set to a Sharded Cluster
=======================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: twocols

This page outlines the procedure to synchronize data from a replica set to a 
sharded cluster using :program:`mongosync`. 

> ``mongosync`` doesn't support sync from a sharded cluster to a replica set.

Reviewer: [suggestion] I would suggest putting this line in a note!

Author: Adding note now!

To perform a sync between two replica sets, see :ref:`c2c-quickstart`.

Before You Begin
----------------

Before you begin your sync:

- Define your source replica set.
- Define your destination sharded cluster.

> - Define an administrative user that has permissions in both clusters.

Reviewer: I really like this clarification!

You must also download ``mongosync``. To learn more, see 
the :ref:`mongosync setup <c2c-quickstart-setup>` instructions.

Steps 
-----

.. procedure::
   :style: normal

   .. step:: Connect the replica set and sharded cluster. 

>       ``mongosync`` must create an initial connection to the source and 
>       destination clusters before you can start to sync data.

Reviewer: [suggestion] I really like how you clarified what the source and destination must be in the "Before You Begin" section - I would suggest doing the same here

```suggestion
      ``mongosync`` must create an initial connection to the source replica set and 
      destination sharded clusters before you can start to sync data.
```

      replica set and the destination sharded cluster before you can start to 
      sync data. 

      To connect ``cluster0`` to ``cluster1`` with ``mongosync``, enter the
      following command:

      .. include:: /includes/example-connection-string-general.rst

      Follow the :ref:`connection instructions <c2c-conn-top-level>` 
      for your cluster architecture to format your connection strings and 
      connect to the :binary:`mongos` instance in your cluster. 

   .. step:: Start synchronization

       Call the :ref:`start <c2c-api-start>` endpoint to initiate data 
       synchronization. 
       
       To sync from a replica set to a sharded cluster, set the 
       ``sharding`` option for the ``start`` command to shard collections on 
       the destination cluster. For more information, see 
       :ref:`Sharding Parameters <c2c-api-start-sharding>`.

       Use the ``sharding.shardingEntries`` parameter to specify the collections 
       to shard. Collections that you do not list in this array replicate as 
       unsharded. For more information, see 
       :ref:`Shard Replica Sets <c2c-shard-replica-sets>`.

> 

Author: Good idea! I added "To choose a shard key, see :ref:`sharding-shard-key-selection`."

       The following example starts a sync from a replica set to a sharded
       cluster: 

       Request
       ~~~~~~~

       .. literalinclude:: /includes/api/requests/start-rs-shard.sh

       Response
       ~~~~~~~~

       .. literalinclude:: /includes/api/responses/success.json
          :language: json
          :copyable: false

Next Steps
----------

You can finalize a migration and transfer your application workload from the 
source to the destination cluster using the ``mongosync`` 
:ref:`cutover process <c2c-cutover-process>`.

Limitations
-----------

- You cannot use the :ref:`reverse <c2c-api-reverse>` endpoint between a 
  replica set and a sharded cluster. 

.. include:: /includes/limitations-rs-to-sharded.rst

- If the source replica set has :ref:`arbiters <c2c-faq-arbiters>`, the source 
  replica set must have more than 2 non-arbiter nodes and you must sync from a 
  non-arbiter node. 

Learn More
----------

> - :ref:`c2c-quickstart`.

Author: Thanks, good catch!

.. _c2c-rs-to-sharded:

=======================================
Sync a Replica Set to a Sharded Cluster
=======================================

> .. default-domain:: mongodb

Reviewer 2: FYI we dont need the default-domain directives anymore, these were leftover from the legacy built tools and can be safely removed from all pages.

```suggestion

```

Author: Good to know, just deleted from this page.

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: twocols

This page outlines the procedure to synchronize data from a replica set to a 
sharded cluster using :program:`mongosync`. 

.. note::

>    ``mongosync`` doesn't support sync from a sharded cluster to a replica set.

Reviewer 2: ```suggestion
   ``mongosync`` doesn't support syncing from a sharded cluster to a replica set.
```

Author: Changing now.

> To perform a sync between two replica sets, see :ref:`c2c-quickstart`.

Reviewer 2: Consider moving this to the Learn More section

```suggestion
To perform a sync between two replica sets, see :ref:`c2c-quickstart`.
```

Author: Good idea, I also don't think this sentence needs to be on this page when it is already on the topology landing page.

Before You Begin
----------------

> Before you begin your sync:

Reviewer 2: I think you can omit this blub, because of the ByB header. Also is it possible to add links to each of the bullets such as Define your :ref:`source replica set <my_link>`
```suggestion

```

Author: I removed the blurb and added links!

- Define your source replica set.
- Define your destination sharded cluster.
- Define an administrative user that has permissions in both clusters.

> You must also download ``mongosync``. To learn more, see 

Reviewer 2: Move this install above all other bullets in the ByB section.

```suggestion
- You must download and install ``mongosync``. To learn more, see 
```

Author: Changing now.

the :ref:`mongosync setup <c2c-quickstart-setup>` instructions.

Steps 
-----

.. procedure::
   :style: normal

>    .. step:: Connect the replica set and sharded cluster. 

Reviewer 2: No punctuation for step titles, this applies globally to this edit. 

```suggestion
   .. step:: Connect the replica set and sharded cluster
```

Author: Good catch, I will keep that in mind from now on!

      ``mongosync`` must create an initial connection between the source 
      replica set and the destination sharded cluster before you can start to 
      sync data. 

      To connect ``cluster0`` to ``cluster1`` with ``mongosync``, enter the

>       following command on one line:

Reviewer 2: I might remove the on one line here, its slightly confusing because the code example has `\` on multiple lines. If relevant, you can add a bullet in the AtT section to the effect:

```
- mongosync commands must be entered on a single line or have line terminated slashes (`\`). 
```
```suggestion
      following command:
```

       to shard. Collections that you do not list in this array replicate as 
       unsharded. For more information, see 
       :ref:`Shard Replica Sets <c2c-shard-replica-sets>`.

       The following example starts a sync from a replica set to a sharded
       cluster: 

       Request
       ~~~~~~~

       .. literalinclude:: /includes/api/requests/start-rs-shard.sh

       Response
       ~~~~~~~~

       .. literalinclude:: /includes/api/responses/success.json
          :language: json
          :copyable: false

Next Steps
----------

You can finalize a migration and transfer your application workload from the 
source to the destination cluster using the ``mongosync`` 
:ref:`cutover process <c2c-cutover-process>`.

Limitations
-----------

- You cannot use the :ref:`reverse <c2c-api-reverse>` endpoint between a 
  replica set and a sharded cluster. 

.. include:: /includes/limitations-rs-to-sharded.rst

- If the source replica set has :ref:`arbiters <c2c-faq-arbiters>`, the source 
  replica set must have more than 2 non-arbiter nodes and you must sync from a 
  non-arbiter node. 

Learn More
----------

> To learn more about ``mongosync``, see:

Reviewer 2: Typically we just have bullet points for the Learn More section

```suggestion

```

Author: Deleting now!
